src/node/grammar/generateParser.js

```diff
@@ -20,7 +20,7 @@ fs.readFile('./src/node/grammar/queryGrammar.pegjs', 'utf8', (err, data) => {
 
   fs.mkdir(
     './cache/advancedSearchQuery',
-    { recursive: true, mode: 0777 },
+    {recursive: true, mode: 0700},
```

Comment:
advancedSearchQuery folder is built by root, so it should be 755, or it breaks the search page because nginx user won't get access!